### PR TITLE
Unpin all messages when help channel goes dormant

### PR DESCRIPTION
**Pull request by dementati**
As described in #1903, unpinning a help channel question will occasionally fail when a channel goes dormant. This issue was previously fixed in #1909 by querying and unpinning all pinned messages in the help channel when it moves from dormant to available. This likely fixes the issue, but the current unpinning solution is still unnecessarily complex.

Unpinning the question when the channel goes dormant is currently solved by storing the question message ID when the channel is first claimed and retrieving it from there to be unpinned at the appropriate time. An alternative and simpler solution, implemented in this PR, is to query all pinned messages and unpin those found, like what happens when the channel becomes available. This allows us to get rid of the message cache with no known disadvantages.

### CODE REVIEW

**Comment by MarkKoz** [comment]
If they're unpinned when the channel becomes available, then why even bother unpinning when it goes dormant?

> This allows us to get rid of the message cache with no known disadvantages.

Getting pinned messages requires an API call to Discord. With the cache, this is avoided. That is basically the only advantage. That being said, this sort of optimisation is not necessary; it was nice to have but it's not worth keeping if it's causing consistency issues.

**Comment by dementati** [comment]
> If they're unpinned when the channel becomes available, then why even bother unpinning when it goes dormant?

Apparently, it's the API itself that bugs out occasionally, and making two passes will decrease the likelihood that there are pins remaining by the time the channel becomes available.

> Getting pinned messages requires an API call to Discord. With the cache, this is avoided. That is basically the only advantage.

That is true.

> it was nice to have but it's not worth keeping if it's causing consistency issues.

As stated above, it's allegedly the API and not the cache that caused the issue. I just prefer the solution in this PR because it reduces code complexity.

**Review by Bluenix2 — APPROVED** [review]
Perfect, works much better than the previous behaviour! Another bug squashed..
image

File: bot/exts/help_channels/_message.py [hunk 1]
```diff
@@ -174,8 +174,7 @@ async def notify(channel: discord.TextChannel, last_notification: t.Optional[Arr
 
 async def pin(message: discord.Message) -> None:
     """Pin an initial question `message` and store it in a cache."""
```
Comment (onerandomusername) [inline bot/exts/help_channels/_message.py:176]: No longer storing to a cache.

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: bot/exts/help_channels/_message.py (hunk 1 above)
Comment (MarkKoz) [inline bot/exts/help_channels/_message.py:176]:
```suggestion
    """Pin an initial question `message`."""
```

File: bot/exts/help_channels/_message.py [hunk 2]
```diff
@@ -201,11 +200,8 @@ async def send_available_message(channel: discord.TextChannel) -> None:
 
 async def unpin(channel: discord.TextChannel) -> None:
     """Unpin the initial question message sent in `channel`."""
```
Comment (MarkKoz) [inline bot/exts/help_channels/_message.py:202]:
```suggestion
    """Unpin all messages sent in `channel`."""
```

**Review by MarkKoz — CHANGES_REQUESTED** [review]

**Review by MarkKoz — COMMENTED** [review]
This will allow a `discord.Message` to be passed to `pin_wrapper` instead of a message ID. Thus, the wrapper will be able to call the `discord.Message.pin` and `discord.Message.unpin` functions rather than using the undocumented methods on `HTTPClient`. Would you mind updating the wrapper function too @dementati?

**Comment by Xithrius** [comment]
@dementati What's your status on this PR?

**Review by MarkKoz — APPROVED** [review]

**Review by ChrisLovering — APPROVED** [review]

**Pull request merged**